test/kt/CMakeLists.txt

```
@@ -124,7 +124,49 @@ if (ONEDPL_TEST_ENABLE_KT_ESIMD)
     _generate_esimd_sort_test("256" "32" "double" "" 1000) # segfault
 endif()
 
+function (_generate_scan_test _data_per_work_item _work_group_size _type)
+
+    if ((NOT TARGET build-scan-tests) AND (NOT TARGET run-scan-tests))
+        add_custom_target(build-scan-tests COMMENT "Build all scan kernel template tests")
+        add_custom_target(run-scan-tests
+            COMMAND "${CMAKE_CTEST_COMMAND}" -R "^run-scan-tests$" --output-on-failure --no-label-summary
+            DEPENDS build-scan-tests
+            COMMENT "Build and run all scan kernel template tests")
+    endif()
+
+    string(REPLACE "_t" "" _type_short ${_type})
+    set(_target_name "single_pass_scan_dpwi${_data_per_work_item}_wgs${_work_group_size}_${_type_short}")
+    set(_test_path "single_pass_scan.cpp")
+
+    #_generate_test_randomly(${_target_name} ${_test_path} ${_probability_permille})
+    _generate_test(${_target_name} ${_test_path})
+    if(TARGET ${_target_name})
+        add_dependencies(build-scan-tests ${_target_name})
+        add_dependencies(run-scan-tests ${_target_name})
+
+        target_compile_definitions(${_target_name} PRIVATE TEST_DATA_PER_WORK_ITEM=${_data_per_work_item})
+        target_compile_definitions(${_target_name} PRIVATE TEST_WORK_GROUP_SIZE=${_work_group_size})
+        target_compile_definitions(${_target_name} PRIVATE TEST_TYPE=${_type})
+    endif()
+
+endfunction()
+
+function(_generate_scan_tests)
+    set(_data_per_work_item_all "8")
+    set(_work_group_size_all "256")
```

Basing on discussion on the main PR, we can expand the list to:

```suggestion
    set(_data_per_work_item_all "1" "2" "4" "8" "16" "32")
    set(_work_group_size_all "64" "128" "256" "512" "1024")
```

---

Fixed